apps/services/license-api/src/app/modules/license/clients/firearmLicense/firearmLicenseApiClient.config.ts

```diff
@@ -0,0 +1,18 @@
+import { defineConfig } from '@island.is/nest/config'
+import { SmartSolutionsApiConfigSchema as schema } from '@island.is/clients/smartsolutions'
+import * as z from 'zod'
+
+export const FirearmLicenseApiClientConfig = defineConfig<
```

Comment:
It is tempting to create some sort factory function to create config definitions to avoid code duplication.

```
// in some other file
export const createClientConfig = (licenseId: string, organisation: string) => {
  return defineConfig<
  z.infer<typeof schema>
>({
  name: `${licenseId}LicenseApiClientConfig`,
  schema,
  load: (env) => ({
    apiKey: env.required(`${organisation}_PKPASS_API_KEY`, ''),
    apiUrl: env.required(
      'SMART_SOLUTIONS_API_URL',
      'https://smartpages-api-dev.smartsolutions.is/graphql',
    ),
    passTemplateId: env.required(`${licenseId.toUpperCase()}_LICENSE_PASS_TEMPLATE_ID`, ''),
  }),
})

export const FirearmLicenseApiClientConfig = createClientConfig('Firearm', 'RLS')
```

Reply:
good shout

Reply:
i'll probably handle this in another ongoing [PR](https://github.com/island-is/island.is/pull/9718), which creates a client factory that the license-api and license-service will use.

Reply:
Sounds good